ui/src/app/components/status-alert-banner-maybe.tsx

```diff
@@ -4,73 +4,77 @@ import { useEffect, useState } from 'react';
 import { StatusAlert, StatusAlertLocation } from 'generated/fetch';
 
 import { statusAlertApi } from 'app/services/swagger-fetch-clients';
-import { firstPartyCookiesEnabled } from 'app/utils/cookies';
+import {
+  isDismissed,
+  saveDismissedMessage,
+} from 'app/utils/dismissed-messages';
 
 import { Button } from './buttons';
-import { ToastBanner, ToastType } from './toast-banner';
-
-const STATUS_ALERT_COOKIE_KEY = 'status-alert-banner-dismissed';
-const INITIAL_STATUS_ALERT: StatusAlert = {
-  statusAlertId: 0,
-  title: '',
-  message: '',
-  link: '',
-};
-
-const shouldShowStatusAlert = (statusAlert: StatusAlert) => {
-  const { statusAlertId, message } = statusAlert;
-  if (firstPartyCookiesEnabled()) {
-    const cookie = localStorage.getItem(STATUS_ALERT_COOKIE_KEY);
-    return (!cookie || (cookie && cookie !== `${statusAlertId}`)) && !!message;
-  } else {
-    return !!message;
-  }
-};
+import { MultiToastBanner } from './multi-toast-banner';
+import { MultiToastMessage } from './multi-toast-message.model';
+import { ToastType } from './toast-banner';
 
 export const StatusAlertBannerMaybe = () => {
-  const [showStatusAlert, setShowStatusAlert] = useState(false);
-  const [statusAlertDetails, setStatusAlertDetails] =
-    useState(INITIAL_STATUS_ALERT);
+  const [alertMessages, setAlertMessages] = useState<MultiToastMessage[]>([]);
 
   useEffect(() => {
-    const getAlert = async () => {
-      const statusAlert = await statusAlertApi().getStatusAlert();
-      if (statusAlert?.alertLocation === StatusAlertLocation.AFTER_LOGIN) {
-        setShowStatusAlert(shouldShowStatusAlert(statusAlert));
-        setStatusAlertDetails(statusAlert);
+    const getAlerts = async () => {
+      const messages: MultiToastMessage[] = [];
+
+      try {
+        const statusAlerts = await statusAlertApi().getStatusAlerts();
```

Comment:
Could probably filter alerts here in order to avoid the if block below, but I have longer term plans of shifting alerts (and possibly all messaging into a provider), so I do not think it is worth the effort right now.